geps/gep-1619.md

```diff
@@ -74,28 +74,122 @@ While session persistence uses attributes in the application layer, session affi
-| Traefik Proxy 	| Traefik 	| [Cookie-Based](https://doc.traefik.io/traefik/routing/services/#sticky-sessions) 	| No 	| [Docs](https://traefik.io/glossary/what-are-sticky-sessions/) mention consistent hashing, but Traefik [only offers round robin](https://doc.traefik.io/traefik/routing/services/#load-balancing) (not consistent). 	|
-| Java Servlet 	| N/A 	| [Cookie-Based](https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServletRequest.html#getSession--), [URL Encoded](https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServletRequest.html#isRequestedSessionIdFromURL--) 	| N/A 	| Java Servlets don't natively support proxy functions. 	|
-| Apache httpd 	| N/A 	| [Cookie-Based](https://httpd.apache.org/docs/2.4/mod/mod_proxy_balancer.html), [URL Encoded](https://httpd.apache.org/docs/2.4/mod/mod_proxy_balancer.html) 	| N/A 	|  N/A	|
-
+In this section, we will describe how implementations achieve session persistence and affinity, along with a breakdown of related configuration options. Input from implementations is appreciated to complete this information.
+
+In the following tables, we will example two types of APIs:
+1. Dataplane APIs
+2. Implementation APIs
+
+Generally, the implementation API programs the dataplane API; however these two are not always clearly separated. The two types of APIs can use different API structures for configuring the same feature. Examining the dataplane APIs helps to remove the layer of API abstraction that implementations provide. Removing this layer avoids situations where implementations don’t fully implement all capabilities of a dataplane API or obfuscate certain configuration around session persistence. On the other hand, examining implementation APIs provides valuable data points in what implementations are interested in configuring.
+
+**Session Persistence**
+
+| **Technology** 	| **Technology Type** 	| **Session Persistence Type** 	| **Configuration Options** 	| **Configuration Association (Global, Gateway, Route, or Backends)** 	| **Notes** 	|
+|---	|---	|---	|---	|---	|---	|
+| Envoy 	| Dataplane 	| [Cookie-Based](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/http/stateful_session/cookie/v3/cookie.proto) 	| Name=name<br>Path=path<br>TTL=duration 	| [HttpConnectionManager](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto) (Route) 	| Envoy does not natively support cookie modification, but Contour has implemented it via Lua ([reference](https://github.com/projectcontour/contour/blob/main/design/cookie-rewrite-design.md)). Lua scripts can be added per route in Envoy via the [Lua envoy filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter) and can modify. [Related Issue](https://github.com/envoyproxy/envoy/issues/15612). 	|
```

Comment:
Not sure what you mean by "Envoy does not natively support cookie modification" but according to this code https://github.com/envoyproxy/envoy/blob/main/source/extensions/http/stateful_session/cookie/cookie.cc#L11 Envoy does set a generated cookie in the response.